### PR TITLE
Fix weird issues with rounding 

### PR DESCRIPTION
**Pull request by walmat**
https://rainbowhaus.slack.com/archives/C04F08AEJAZ/p1717686637601269

Also adding some test suites for the NiceIncrementFormatter

### CODE REVIEW

File: src/__swaps__/utils/swaps.ts [hunk 1]
```diff
@@ -295,37 +291,59 @@ export function niceIncrementFormatter({
...
+      precisionAdjustment: precisionBasedOffMagnitude(amount),
+      isStablecoin,
     });
-  if (percentageToSwap === 1)
+  }
+  if (percentageToSwap === 1) {
     return valueBasedDecimalFormatter({
       amount: inputAssetBalance,
       usdTokenPrice: inputAssetUsdPrice,
-      roundingMode: 'up',
```
Comment (walmat) [inline src/__swaps__/utils/swaps.ts:326]: if we're at 100% percent we want to round normally, not exceed their balance

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 2]
```diff
@@ -119,10 +123,13 @@ export function useSwapInputsController({
       return addCommasToNumber(inputValues.value.inputAmount, '0');
     }
 
+    const assetBalanceDisplay = internalSelectedInputAsset.value.balance.display ?? '';
```
Comment (christianbaroni) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:126]: Why the `display` amount? Also should the fallback be 0?

Comment (walmat) [reply src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:126]: This was my attempt to get as close to what we display in the balance badge for selling 100%

File: config/test/jest-setup.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 /* eslint-disable no-nested-ternary */
+import { setUpTests } from 'react-native-reanimated';
```
Comment (walmat) [inline config/test/jest-setup.js:2]:
```suggestion
```

File: config/test/jest-setup.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
...
 import '../../src/languages';
 
+setUpTests();
+
```
Comment (walmat) [inline config/test/jest-setup.js:6]:
```suggestion
```

File: config/test/jest-setup.js [hunk 5]
```diff
@@ -27,6 +30,8 @@ jest.mock('@sentry/react-native', () => ({
   captureException: () => null,
 }));
 
+jest.mock('react-native-reanimated', () => require('react-native-reanimated/mock'));
+
```
Comment (walmat) [inline config/test/jest-setup.js:34]:
```suggestion
```

File: jest.config.js [hunk 6]
```diff
@@ -16,7 +16,7 @@ module.exports = {
     '\\.tsx?$': 'ts-jest',
   },
   transformIgnorePatterns: [
-    'node_modules/(?!((jest-)?react-native|react-native-keyboard-area|imgix-core-js|react-native-payments|@react-native-firebase|@react-native(-community)?)/)',
+    'node_modules/(?!((jest-)?react-native|react-native-reanimated|react-native-keyboard-area|imgix-core-js|react-native-payments|@react-native-firebase|@react-native(-community)?)/)',
```
Comment (walmat) [inline jest.config.js:19]:
```suggestion
    'node_modules/(?!((jest-)?react-native|react-native-keyboard-area|imgix-core-js|react-native-payments|@react-native-firebase|@react-native(-community)?)/)',
```

File: src/__swaps__/utils/swaps.ts (hunk 1 above)
Comment (brunobar79) [reply src/__swaps__/utils/swaps.ts:326]: Good catch!

**Review by brunobar79 — COMMENTED** [review]
Added 2 minor comments otherwise looks good to me! Ping me and I'll approve after fixing those

**Review by brunobar79 — APPROVED** [review]
LGTM!

**Pull request merged**